### PR TITLE
Feature 1123 ddb update

### PR DESCRIPTION
**Pull request by MugPand**
# DynamoDB & Terraform Update

Github Issue Number Here: GitHub #1123
**What user problem are we solving?**
The current DynamoDB tables and Terraform script reflect an older version of infra-configurations. 

**What solution does this PR provide?**
Updated DynamoDB tables and Terraform script to reflect new changes proposed by the following diagram: **Also added user_id as GSI to ModelTable**.
image

**Testing Methodology**
> How did you test your changes and verify that existing functionality is not broken
Ran terraform file configurations and validated created DynamoDB Tables on AWS.

**Any other considerations**

### CODE REVIEW

**Comment by farisdurrani** [comment]
Delete the `*.tfstate*` files. Ignore the Sonarcloud warning

**Comment by karkir0003** [comment]
> Delete the `*.tfstate*` files. Ignore the Sonarcloud warning

@farisdurrani should this be gitignored?

File: serverless/packages/functions/src/trainspace/create_trainspace.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
+import { APIGatewayProxyHandlerV2 } from "aws-lambda";
```
Comment (karkir0003) [inline serverless/packages/functions/src/trainspace/create_trainspace.ts:1]: @MugPand i think the api logic and terraform stuff is mixed int the same PR

**Comment by farisdurrani** [comment]
Yes. Deleted and gitignored

**Pull request closed without merging**